sinks/gcm/driver.go

```diff
@@ -83,67 +85,71 @@ type metricDescriptor struct {
 
 const maxNumLabels = 10
 
-// Substitutes any generic description with GCM-specific descriptions
-func getDescription(metric sink_api.MetricDescriptor) string {
-	descriptions := []struct {
-		// Metric names for this description.
-		name string
-
-		// Description to use
-		description string
-	}{
-		{
-			name:        "uptime",
-			description: "Rate of change of time since start in seconds per second",
-		},
-		{
-			name:        "cpu/usage",
-			description: "Rate of total CPU usage in millicores per second",
-		},
-		{
-			name:        "memory/page_faults",
-			description: "Rate of major page faults in counts per second",
-		},
-		{
-			name:        "network/rx",
-			description: "Rate of bytes received over the network in bytes per second",
-		},
-		{
-			name:        "network/rx_errors",
-			description: "Rate of errors sending over the network in errors per second",
-		},
-		{
-			name:        "network/tx",
-			description: "Rate of bytes transmitted over the network in bytes per second",
-		},
-		{
-			name:        "network/tx_errors",
-			description: "Rate of errors transmitting over the network in errors per second",
-		},
-	}
-
-	// Replac the description if we have an alternate one.
-	for _, desc := range descriptions {
-		if metric.Name == desc.name {
-			return desc.description
-		}
-	}
-
-	return metric.Description
-}
-
 // Map of metric name to translation function.
 var translationFuncs = map[string]func(float64) float64{
-	"uptime": func(value float64) float64 {
+	"uptime_rate": func(value float64) float64 {
 		// Convert from milliseconds to seconds.
 		return value / 1000
 	},
-	"cpu/usage": func(value float64) float64 {
+	"cpu/usage_rate": func(value float64) float64 {
 		// Convert from billionths of a core to millicores.
 		return value / 1000000
 	},
 }
 
+type listMetricsResponse struct {
+	Metrics []metricDescriptor `json:"metrics,omitempty"`
+}
+
+func (self *gcmSink) listMetrics() error {
+	response := listMetricsResponse{}
+	err := sendRequest("GET", fmt.Sprintf("https://www.googleapis.com/cloudmonitoring/v2beta2/projects/%s/metricDescriptors", self.project), self.token, nil, &response)
```

Comment:
I think we can move the URL to a separate function (or a const) since we use it in multiple places